Spatial.gd

```diff
@@ -692,25 +692,32 @@ func gen_adjustment_nodes():
 				continue
 			var new_gizmo = gizmo_scene.instance()
 			new_gizmo.translation = gizmo_position
-			new_gizmo.link_point("path", route, path2d, i)
-			new_gizmo.connect("selected", self, "on_gizmo_selected")
-			new_gizmo.connect("deselected", self, "on_gizmo_deselected")
+			new_gizmo.connect("selected", self, "on_gizmo_selected", [route, "path", path2d, i])
+			new_gizmo.connect("deselected", self, "on_gizmo_deselected", [route, "path", path2d, i])
+			new_gizmo.connect("updated", route, "update_point_poistion", [i])
+			new_gizmo.connect("updated", path2d, "update_point_poistion", [i])
 			$Gizmos.add_child(new_gizmo)
 	for icon in category3d.icons:
 		var new_gizmo = gizmo_scene.instance()
 		new_gizmo.translation = icon.translation
-		new_gizmo.link_point("icon", icon)
-		new_gizmo.connect("selected", self, "on_gizmo_selected")
-		new_gizmo.connect("deselected", self, "on_gizmo_deselected")
+		new_gizmo.connect("selected", self, "on_gizmo_selected", [icon, "icon"])
+		new_gizmo.connect("deselected", self, "on_gizmo_deselected", [icon, "icon"])
```

Comment:
You have a point. To address this, I'm going to split the function `on_gizmo_selected` into unique functions for the type of node. This will eliminate using the optional variables and make it clear what buttons should be active for each node type.